packages/library/src/commonMain/kotlin/io/realm/MutableRealm.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import io.realm.internal.RealmModelInternal
+import io.realm.internal.unmanage
+import io.realm.interop.NativePointer
+import io.realm.interop.RealmInterop
+import kotlin.reflect.KClass
+
+/**
+ * This class represents the writeable state of a Realm file. The only way to modify data in this file is through
+ * instances of this class. These are provided and managed automatically through either [Realm.write] or
+ * [Realm.writeBlocking].
+ */
+class MutableRealm : BaseRealm {
+
+    companion object {
+        // FIXME EVALUATE Should this be on RealmModel instead?
+        fun <T : RealmObject> delete(obj: T) {
```

Comment:
This is calleable from an arbitrary context. See comment on the `RealmObject.delete` extension method.